### ISSUE TITLE
Release readxl 1.4.0

### ISSUE THREAD

**Issue by jennybc** [tidyverse/readxl#691]
Prepare for release:

* [x] `git pull`
* [x] Check [current CRAN check results](https://cran.rstudio.org/web/checks/check_results_readxl.html)
  - There's the usual NOTE: `'LazyData' is specified without a 'data' directory` but I've already fixed in dev.
  - Also the NOTE: `installed size is 5.3Mb, sub-directories of 1Mb or more: libs 4.0Mb` but that's been true for a while and we can't really do anything about it. Flavor: [r-release-macos-arm64](https://www.r-project.org/nosvn/R.check/r-release-macos-arm64/readxl-00check.html).
* [x] [Polish NEWS](https://style.tidyverse.org/news.html#news-release)
* [x] `devtools::build_readme()`
* [x] `urlchecker::url_check()`
* [x] `devtools::check(remote = TRUE, manual = TRUE)`
* [x] `devtools::check_win_devel()`
* [x] `rhub::check_for_cran()`
* [x] `rhub::check(platform = 'ubuntu-rchk')`
* [x] `rhub::check_with_sanitizers()`
* [x] `revdepcheck::cloud_check()`
* [x] Update `cran-comments.md`
* [x] `git push`
* [x] Draft blog post
* [ ] ~Slack link to draft blog in #open-source-comms~ *this is not that kind of release*

Submit to CRAN:

* [x] `usethis::use_version('minor')`
* [x] `devtools::submit_cran()`
* [x] Approve email

Wait for CRAN...

* [x] Accepted :tada:
* [x] `git push`
* [x] `usethis::use_github_release()`
* [x] `usethis::use_dev_version()`
* [x] `git push`
* [x] Finish blog post
* [x] Tweet
* [x] Add link to blog post in pkgdown news menu

**Comment by jennybc** [comment]
Seen with `devtools::check_win_devel()`, specifically `R Under development (unstable) (2022-03-23 r81968 ucrt)`

```
win-builder 2 tests fail
* using R Under development (unstable) (2022-03-23 r81968 ucrt)
* using platform: x86_64-w64-mingw32 (64-bit)
* using session charset: UTF-8

Failure (test-dates.R:16:3): date subsecond rounding works 
  as.character(df$dttm) (`actual`) not identical to df$dttm_string (`expected`).
  
      actual       | expected                 
  [1] "2016-04-28" - "2016-04-28 11:30:00" [1]
  [2] "2016-03-23" - "2016-03-23 09:50:00" [2]
  [3] "2017-01-02" - "2017-01-02 11:30:00" [3]
  [4] "2017-01-03" - "2017-01-03 11:30:00" [4]
  
Failure (test-dates.R:35:3): we get correct dates prior to March 1, 1900, in 1900 date system [xlsx]
  df$dttm[!leap_day] (`actual`) not identical to dttms[!leap_day] (`expected`).
  
      actual       | expected                 
  [1] "1900-01-01" - "1900-01-01 08:00:00" [1]
  [2] "1900-01-02" - "1900-01-02 08:00:00" [2]
  [3] "1900-02-28" - "1900-02-28 08:00:00" [3]
  [4] "1900-03-01" - "1900-03-01 08:00:00" [4]
  [5] "1903-12-31" - "1903-12-31 08:00:00" [5]
  [6] "1904-01-01" - "1904-01-01 08:00:00" [6]
  
  [ FAIL 2 | WARN 0 | SKIP 6 | PASS 367 ]
```

https://win-builder.r-project.org/IC2JGB7msk7b/00check.log

**Comment by jennybc** [comment]
However, on r-hub, with the same version of r-devel, there are no test failures:

r-hub readxl 1.3.1.9000: OK
Build ID:	readxl_1.3.1.9000.tar.gz-5f07cea3e02a414b8ecf7be299eaac00
Platform:	Windows Server 2022, R-devel, 64 bit
* using R Under development (unstable) (2022-03-23 r81968 ucrt)
* using platform: x86_64-w64-mingw32 (64-bit)
* using session charset: UTF-8

*
https://builder.r-hub.io/status/readxl_1.3.1.9000.tar.gz-5f07cea3e02a414b8ecf7be299eaac00

**Comment by jennybc** [comment]
On a local Windows 10 VM, with:

``` r
> R.version.string
[1] "R Under development (unstable) (2022-03-23 r81968 ucrt)"
> getOption("encoding")
[1] "native.enc"
> l10n_info()
$MBCS
[1] TRUE

$`UTF-8`
[1] TRUE

$`Latin-1`
[1] FALSE

$codepage
[1] 65001

$system.codepage
[1] 65001
```

The 2 tests in `test-dates.R` that fail on `check_win_devel()` and pass on r-hub ALSO pass.

Interestingly, two *different* tests fail, both seemingly with some sort of UTF-8 angle:

https://github.com/tidyverse/readxl/blob/6896d8dd3fbe470542c34f7edf87c8995561064e/tests/testthat/test-compatibility.R#L71-L82

https://github.com/tidyverse/readxl/blob/6896d8dd3fbe470542c34f7edf87c8995561064e/tests/testthat/test-encoding.R#L7-L10

I do not see these failures on win-builder or r-hub, so I'm chalking this up to user error or sub-optimal configuration on my VM. Or infelicity with this particular r-devel snapshot.

**Comment by jennybc** [comment]
I am unsure how to interpret the result from rchk.

But this is typical of the output:

```
Function SheetView<Xls>::readCols(cpp11::r_vector<cpp11::r_string>, std::__1::vector<ColType, std::__1::allocator<ColType> > const&, StringSet const&, bool, bool)
  [UP] ignoring variable <unnamed var:   %26 = alloca %struct.SEXPREC*, align 8> as it has address taken, results will be incomplete
```

which is (sort of?) explained here:

https://github.com/kalibera/rchk/issues/13

I'm interpreting this as "no concrete problems found".

**Comment by jennybc** [comment]
As expected / feared, the `check_win_devel()` result repeated itself upon CRAN submission:

https://win-builder.r-project.org/incoming_pretest/readxl_1.4.0_20220325_001111/Windows/00check.log

I described my efforts to replicate and linked to the passing r-hub check. It will be interesting to see what they say.

**Comment by jennybc** [comment]
@gaborcsardi was able to replicate the 2 test failures seen with `devtools::check_win_devel()` by setting the system language to German. This appears to be related to patches to locale handling in libxls, which were crafted before the announcement that R 4.2 will use UCRT on Windows.

**Comment by jennybc** [comment]
I've now replicated the failure in my own VM, with display language set to German.

The fix in #694 looks promising.
